web/src/components/dashboard/QuickActionTile.tsx

```diff
@@ -0,0 +1,60 @@
+import { ROUTES } from "@/lib/domain-logic/routes";
+import { QuickActionLink, QuickActionTask } from "@/lib/types/types";
+import analytics from "@/lib/utils/analytics";
+import { useRouter } from "next/router";
+import { ReactElement, ReactNode } from "react";
+
+interface LinkProps {
+  type: "link";
+  quickAction: QuickActionLink;
+}
+
+interface TaskProps {
+  type: "task";
+  quickAction: QuickActionTask;
+}
+
+export const QuickActionTile = (props: LinkProps | TaskProps): ReactElement => {
```

Comment:
One of the topics we've covered in our tech video sessions is "making impossible types impossible". This is a [discriminated union](https://basarat.gitbook.io/typescript/type-system/discriminated-unions). Leveraging this lets us consolidate 2 separate use cases of this component without having to handle for extensive conditional logic, or as existed previously, 2 separate but similar component definitions.

In this specific instance you could argue that "link" and "task" could be combined further given that `urlSlug` and `externalRoute` are similar enough data, but I'm intentionally opting to keep that separate.